jwst/undersampling_correction/undersampling_correction.py

```diff
@@ -0,0 +1,76 @@
+
+#  Module for undersampling correction
+#
+import logging
+import numpy as np
+
+from ..datamodels import dqflags
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def undersampling_correction(input_model, signal_threshold):
+    """
+    Short Summary
+    -------------
+    Correct for undersampling.
+
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.RampModel`
+        The input science data to be corrected
+
+    signal_threshold : float
+        Science value above which a group will be flagged as DO_NOT_USE
+
+    Returns
+    -------
+    output_model : `~jwst.datamodels.RampModel`
+        Data model with undersampling_correction applied; add DO_NOT_USE flag
+        to groups exceeding signal_threshold
+    """
+    data = input_model.data
+    gdq = input_model.groupdq
+
+    # Create the output model as a copy of the input
+    output_model = input_model.copy()
+
+    log.info('Using signal_threshold: %s', signal_threshold)
```

Comment:
signal_threshold is a floating-point value, so does the use of "%s" formatting work properly for it?

Reply:
Works as is, but changed for clarity.